### PR TITLE
Add partial empty state for project search

### LINKED ISSUE #2588: Explorer: add partial empty state for project search
Follow-up ticket to: https://github.com/gitcoinco/grants-stack/issues/2488 , per conversations with @boudra [here](https://discord.com/channels/562828676480237578/1174027891806322768/1174064873861287946). See past comment [here](https://github.com/gitcoinco/grants-stack/pull/2560#pullrequestreview-1723467800)) for more context as well. 

**User Story:**
As a donor 
I want to know if the project I'm looking for is not there 
So that I don't waste time looking for a non-existent project

**Acceptance Criteria**

GIVEN that I enter keywords that have no fulltext match but some semantic search matches 
WHEN I look at the page 
THEN I see a partial empty state with copy that says "Your search did not match any projects. Try again or feel free to browse through projects similar to your search" 
AND a list of semantic search results 

GIVEN that I enter keywords that have no fulltext or semantic search matches 
WHEN I look at the page 
THEN I see a full empty state with copy that says "Your search did not match any projects. Try again using different keywords." 
AND no results 

**Product & Design Links:**
* [Figma](https://www.figma.com/file/1uxgU2jb0yyN11SUgxARmF/Explorer-S19?type=design&node-id=792-11219&mode=design&t=X4tN3U0nGAHIKDwt-0)

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by bhargavaparoksham**
## PR checklist

For every PR, make sure that these statements are true:
- [x] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [x] Doesn't contain type casts and non-null assertions.
- [x] Doesn't add `@ts-ignore`.
- [x] Doesn't disable lints.
- [x] Doesn't use `useState` just for computation - use plain variables instead.
- [x] Splits components into pure components that don't depend on external state or hooks.
- [x] Avoid embedding components within other components
- [x] Doesn't propagate optional values without good reason, doesn't mark property values as optional if that doesn't represent reality.
- [x] Doesn't duplicate existing code.
- [x] Parses out-of-domain data - this includes user input, API respones, on-chain data etc.
- [x] Doesn't contain commented out code.
- [x] Doesn't contain skipped or empty tests.
- [x] If this PR adds/updates any feature, it adds/updates its test script

Subjective - at the discretion of the reviewers
- Does things as simply as possible, but not simpler.
- Doesn't reinvent the wheel or create premature abstractions.

##### Description

As a donor
I want to know if the project I'm looking for is not there
So that I don't waste time looking for a non-existent project

##### Refers/Fixes

fixes #2588

### CODE REVIEW

**Review by melissa-neira — APPROVED** [review]
lgtm!

File: packages/grant-explorer/src/features/discovery/ExploreProjectsPage.tsx [hunk 1]
```diff
@@ -103,6 +107,17 @@ export function ExploreProjectsPage(): JSX.Element {
 
   const { projects, add, remove } = useCartStorage();
 
+  useEffect(() => {
```
Comment (boudra) [inline packages/grant-explorer/src/features/discovery/ExploreProjectsPage.tsx:110]: you don't need the useEffect and the useState here, you can compute this on render or in a useMemo

https://react.dev/learn/you-might-not-need-an-effect

Comment (carlbarrdahl) [reply packages/grant-explorer/src/features/discovery/ExploreProjectsPage.tsx:110]: +1

Comment (bhargavaparoksham) [reply packages/grant-explorer/src/features/discovery/ExploreProjectsPage.tsx]: Thanks. Fixed.

**Review by carlbarrdahl — APPROVED** [review]

**Review by vacekj — APPROVED** [review]

**Review by melissa-neira — COMMENTED** [review]
The search works great now, but when I click on collections I see this screen with an empty state. the partial empty state is not supposed to exist for collections. @bhargavaparoksham 
Screenshot 2023-11-16 at 14 36 22

**Comment by bhargavaparoksham** [comment]
@melissa-neira this seems to be a different bug. If you see the text its an empty state message not, partially empty state message.

If the projects are not loading when you click on collections, then its showing a empty state message. Can you raise a different bug ticket?

**Review by michellema1208 — APPROVED** [review]

**Pull request merged**